source/slang/slang-emit-c-like.cpp

```diff
@@ -1236,6 +1236,34 @@ void CLikeSourceEmitter::emitIntrinsicCallExpr(
                 }
                 break;
 
+            case 'S':
```

Comment:
This is fine as a solution, but I'm actually a bit surprised we don't have a `$T` command that works here to look up one of the generic arguments (instead of the value arguments). If we had that as a general-purpose tool we could just use `$T0` in all these cases, because the scalar type to use is one of the arguments to the generic builtin.